assignments/assignment02.rb

```diff
@@ -73,3 +107,156 @@ def pluck(ary, key)
 # - daily balance
 # - summary:
 #   - starting balance, total deposits, total withdrawals, ending balance
+
+
+def create_transaction(date, payee, amount, type)
+  {date: date, payee: payee, amount: amount, type: type}
+end
+
+File.open("bankdata.csv") do |input|
+  ary = input.readlines
+  header = ary.shift
+
+  records = ary.map do |line|
+    fields = line.split ','
+    create_transaction fields[0], fields[1], fields[2], fields[3].chop
+  end
+
+  def extract_withdrawal(records)
+    withdrawals = []
+    records.each do |line|
+      if line[:type] == 'withdrawal'
+        withdrawals << line
+      end
+    end
+    withdrawals
+  end
+
+  def extract_deposit(records)
+    deposits = []
+    records.each do |line|
+      if line[:type] == 'deposit'
+        deposits << line
+      end
+    end
+    deposits
+  end
+
+  #Make a method to go through each line of the input file.
+  #For each day, sum the transactions to get a daily delta
+  #add to the running balance.
+  #day_balance = [{:date, :balance}]
+  def daily_balance(records)
+    day = 1
+    daily_bal = 0
+    day_balance = []
+    while day <= 31
```

Comment:
You could do this whole loop with map to select all transaction for a particular day, and then use reduce to calculate the delta in daily balance